src/main/java/org/codeforamerica/messaging/services/EmailService.java

```diff
@@ -13,16 +15,28 @@ public class EmailService {
     private final MailgunGateway mailgunGateway;
 
     private final EmailMessageRepository emailMessageRepository;
+    private final EmailSubscriptionRepository emailSubscriptionRepository;
 
-    public EmailService(MailgunGateway mailgunGateway, EmailMessageRepository emailMessageRepository) {
+    public EmailService(MailgunGateway mailgunGateway, EmailMessageRepository emailMessageRepository, EmailSubscriptionRepository emailSubscriptionRepository) {
         this.mailgunGateway = mailgunGateway;
         this.emailMessageRepository = emailMessageRepository;
+        this.emailSubscriptionRepository = emailSubscriptionRepository;
     }
 
-    public EmailMessage sendEmailMessage(String to, String body, String subject) {
-        EmailMessage message = mailgunGateway.sendMessage(to, body, subject);
-        message = emailMessageRepository.save(message);
-        log.info("Message sent, Mailgun response: " + message);
+    public EmailMessage sendEmailMessage(String toEmail, String body, String subject) {
+        EmailMessage message = null;
+        if (!unsubscribed(toEmail)) {
+            message = mailgunGateway.sendMessage(toEmail, body, subject);
+            message = emailMessageRepository.save(message);
+            log.info("Message sent, Mailgun response: " + message);
```

Comment:
This response may also have PII, might want to reconsider logging the whole thing